### PR TITLE
Rework and unify gemspecs.

### PR DESCRIPTION
**Pull request by zenspider**
First step is to get them agreeing on dependencies.

There's still some oddities between train vs train-core and
dependencies for AWS and the like, but this brings them in line so
diffing between them is MUCH better.

Next is to gut inspec and push everything over to inspec-core.

Signed-off-by: Ryan Davis <zenspider@chef.io>

### CODE REVIEW

File: inspec.gemspec [hunk 1]
```diff
@@ -11,48 +11,27 @@ Gem::Specification.new do |spec|
...
-  spec.require_paths = ["lib"]
+  # ONLY the aws/azure/gcp files. The rest will come in from inspec-core
+  # the gemspec is necessary for appbundler so don't remove it
+  spec.files =
+    Dir.glob("{{lib,etc}/**/*,inspec.gemspec}")
+      .grep(/aws|azure|gcp|gemspec/)
+      .reject { |f| File.directory?(f) }
 
-  spec.required_ruby_version = ">= 2.4"
+  spec.add_dependency "inspec-core", "= #{Inspec::VERSION}"
```
Comment (tas50) [inline inspec.gemspec:25]: Doesn't this mean we always ship 2 gems?

Comment (tas50) [reply inspec.gemspec:25]: Any reason not to do something like this? https://github.com/chef/chef/blob/master/chef-universal-mingw32.gemspec#L1

File: inspec.gemspec [hunk 2]
```diff
@@ -15,38 +15,15 @@ Gem::Specification.new do |spec|
...
-  spec.add_dependency "tty-prompt", "~> 0.17"
-  spec.add_dependency "tomlrb", "~> 1.2"
-  spec.add_dependency "addressable", "~> 2.4"
-  spec.add_dependency "parslet", "~> 1.5"
-  spec.add_dependency "semverse"
-  spec.add_dependency "htmlentities"
-  spec.add_dependency "multipart-post"
-  spec.add_dependency "term-ansicolor"
+  spec.add_dependency "inspec-core", "= #{Inspec::VERSION}"
+  spec.add_dependency "inspec-bin",  "= #{Inspec::VERSION}"
```
Comment (clintoncwolfe) [inline inspec.gemspec]: We're currently setup to do this dep in the other direction, with `inspec-bin` depending on `inspec`. We need that so the the `inspec` library (which is free to use) may be used by the community without requiring `inspec-bin` (which is license-encumbered). That was part of our commitment to the community in the business model change.

**Review by clintoncwolfe — CHANGES_REQUESTED** [review]
Need to remove the dep on inspec-bin, then should be all set AFAICT.

File: inspec.gemspec (hunk 2 above)
Comment (zenspider) [reply inspec.gemspec]: I thought that was inspec-core now?

Otherwise, what's the point of inspec-core?

File: inspec.gemspec (hunk 1 above)
Comment (zenspider) [reply inspec.gemspec:25]: That's gross... I kinda like it.

Comment (zenspider) [reply inspec.gemspec:25]: Also: 2 gems? We ship 3. 

https://rubygems.org/gems/inspec-bin
https://rubygems.org/gems/inspec-core
https://rubygems.org/gems/inspec

**Review by miah — APPROVED** [review]

File: inspec.gemspec (hunk 1 above)
Comment (clintoncwolfe) [reply inspec.gemspec:25]: We ship 4 gems. There is also inspec-core-bin.

File: inspec.gemspec (hunk 2 above)
Comment (clintoncwolfe) [reply inspec.gemspec]: inspec-core is a lighter-weight version of the library that does not contain anything that needs to be compiled, nor any of the "heavy" plugins such as AWS, Azure, etc. Along with the library, we provide a licensed binstub gem, inspec-core-bin.

**Review by clintoncwolfe — APPROVED** [review]
The linter is unhappy, but looks good now to me otherwise - thanks!

**Pull request merged**